### DISCUSSION TITLE
there should be a state that sets the git remotes on a repo.

### DISCUSSION THREAD

**Discussion by eedgar** [saltstack/salt#67625] [Feature Requests]
There is no prebuilt state to set git remotes.  This means I have to create my own state or I need to use the cmd.run or git module backend.  

git.remote.exists;
    name: upstream
    path: /path/to/repo
    url: git@foo/repo
    
    
there should be an absent version of this as well.
git.remote.absent:
    name: upstream
    path: /path/to/repo

**Comment by OrangeDog** [comment]
The `git.latest` state can be used to add a remote, if you also want to checkout.

> remote='origin'
>
> Git remote to use. If this state needs to clone the repo, it will clone it using this value as the initial remote name. If the repository already exists, and a remote by this name is not present, one will be added.
